profile/store.go

```diff
@@ -99,14 +107,25 @@ func (m *MemStore) SetOwner(own *Profile) error {
 }
 
 // PutProfile adds a peer to this store
-func (m *MemStore) PutProfile(profile *Profile) error {
-	if profile.ID.String() == "" {
+func (m *MemStore) PutProfile(p *Profile) error {
+	if p.ID.String() == "" {
 		return fmt.Errorf("profile.ID is required")
 	}
 
 	m.Lock()
-	m.store[profile.ID] = profile
+	m.store[p.ID] = p
 	m.Unlock()
+
+	if p.PubKey != nil {
+		if err := m.keyStore.AddPubKey(peer.ID(p.GetKeyID()), p.PubKey); err != nil {
```

Comment:
Yeah, I've been fighting it myself, and only reason I carried it is cause I started out with it and then build around it. Since we're sporting our own keybook anyways, I'll clean this all up.